go/roothash/api/commitment/pool.go

```diff
@@ -250,6 +292,42 @@ func (p *Pool) addOpenExecutorCommitment(
 	return nil
 }
 
+// CheckProposerTimeout verifies executor timeout request conditions.
+func (p *Pool) CheckProposerTimeout(
+	ctx context.Context,
+	block *block.Block,
+	sv SignatureVerifier,
+	nl NodeLookup,
+	id signature.PublicKey,
+	round uint64,
+) error {
+	if p.Committee == nil {
+		return ErrNoCommittee
+	}
+	if p.Committee.Kind != scheduler.KindComputeExecutor {
+		return ErrInvalidCommitteeKind
+	}
+
```

Comment:
There is no more round check, it should probably still check if the round matches?

Reply:
Oh yeah of course, i didn't mean to remove that.